barretenberg/cpp/src/barretenberg/vm/avm_trace/avm_alu_trace.cpp

```diff
@@ -431,4 +439,225 @@ FF AvmAluTraceBuilder::op_eq(FF const& a, FF const& b, AvmMemoryTag in_tag, uint
     return res;
 }
 
+/**
+ * @brief This is a helper function that decomposes the input into the various registers of the ALU.
+ *        This additionally increments the counts for the corresponding range lookups entries.
+ * @return A triplet of <alu_u8_r0, alu_u8_r1, alu_u16_reg>
+ */
+template <typename T>
+std::tuple<uint8_t, uint8_t, std::vector<uint16_t>> AvmAluTraceBuilder::to_alu_slice_registers(T a)
+{
+    auto alu_u8_r0 = static_cast<uint8_t>(a);
+    a >>= 8;
+    auto alu_u8_r1 = static_cast<uint8_t>(a);
+    a >>= 8;
+    std::vector<uint16_t> alu_u16_reg{};
```

Comment:
I'm hoping to use this function for all the other range checks in the ALU. So it is templated because we may use other sizes . E.g. in `op_add` we focus on using `uint128_t` so this will save us doing conversions within each function.